<!doctype html>
<html lang="en">

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC" crossorigin="anonymous">
  <link href="{{ url_for('Static', path='/style.css') }}" rel="stylesheet">
  <link rel="stylesheet" href="../Static/style.css"> 
  <title>BBB Data Repository</title>
</head>


<body>
  
    
  <!-- Navbar -->
  <nav class="navbar navbar-expand-lg navbar-dark bg-dark sticky-top">
   <div class="container-fluid">
     <a class="navbar-brand" href=""><h1><p class="text-primary">BBB Data Repository</p></h1></a>
     <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarColor01" aria-controls="navbarColor01" aria-expanded="false" aria-label="Toggle navigation">
       <span class="navbar-toggler-icon"></span>
     </button>
     <div class="collapse navbar-collapse" id="navbarColor01">
       <ul class="navbar-nav me-auto">
         <!-- add navbar items here -->
         <li class="nav-item px-5">
           <a class="nav-link" href="/">Homepage</a>
         </li>
         <li class="nav-item px-5">
           <a class="nav-link active" href="/omics">RNA-Seq
            <span class="visually-hidden">(current)</span>
           </a>
         </li>
         <li class="nav-item px-5">
           <a class="nav-link" href="">Section 2</a>
         </li>
         <li class="nav-item px-5">
           <a class="nav-link" href="">Section 3</a>
         </li>
         <li class="nav-item px-5">
          <a class="nav-link" href="/downloads">Downloads</a>
        </li>
       </ul>
     </div>
   </div>
  </nav>

  <!-- Body container -->
  <div class="container-fluid p-4">
    <!-- Metadata -->
    <div class="row">
      <h1 class="text-primary">Methods</h1>
    </div>
    <br>

    <div class="row">
      <h3 class="text-muted">Raw data, alignment & quantification</h3>
      <p>
        Raw read data was downloaded from the <a class="text-info" href="https://www.ebi.ac.uk/ena/browser/home" target="_blank">European Nucleotide Archive</a> FASP server.
        Reads were aligned to the human genome using <a class="text-info" href="https://github.com/alexdobin/STAR" target="_blank">STAR</a> (release 2.7.0e) with the reference index generated
        from <a class="text-info" href="https://ftp.ensembl.org/pub/release-107/">GRCh38 - Release 107</a> primary assembly file & annotations. Alignments were quantified through
        R using the featureCounts method provided by the <a class="text-info" href="https://bioconductor.org/packages/release/bioc/html/Rsubread.html" target="_blank">Rsubread</a> package (ver. 2.8.2).
      </p>
    </div>
    
    <br>

    <div class="row">
      <h3 class="text-muted">Differential gene expression analyses</h3>
      <p>
        Pairwise differential gene expression analyses between brain endothelial cells (NOT sourced from stem cells) and other tissues were conducted using two approaches.
        First, <a class="text-info" href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html" target="_blank">DESeq2</a> was used with a zero-intercept design which also integrated batch variables 
        (indicated by layout and instrument model in the sample metadata). Model contrasts were extracted for each of the 15 comparisons. A second group of contrasts was obtained only by considering vital tissues (colon, heart, intestine, kidney, liver, lung, vessel).
        Genes with adjusted P < 0.05 and fold change (log2) > 1 were considered significant, which were then ranked based on the number of times they have appeared as significant over 15 or 8 comparisons depending on the run.
      </p>
      <p>
        Using the normalized (median of ratios, see <a class="text-info" href="https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html" target="_blank">here</a>) count matrices, <a class="text-info" href="https://rpubs.com/LiYumei/806213" target="_blank">Wilcoxon rank-sum test over R</a> was conducted 
        with the same two contrast groups to compare for potential false positives from the negative binomial model. Here, genes with FDR corrected P < 0.05 and fold change (log2) > 0 were considered significant, which were again ranked in the previous manner.   
      </p>
      <p>
        The resulting 4 ranked lists were aggregated through <a class="text-info" href="https://cran.r-project.org/web/packages/RobustRankAggreg/index.html" target="_blank">RobustRankAggreg</a> to derive the final list below (with the tag <a class="text-warning" style="text-decoration:none"><strong>All</strong></a>).
        List pairs from the 2 workflows were also aggregated separately (tagged <a class="text-warning" style="text-decoration:none"><strong>DESeq2</strong></a> and <a class="text-warning" style="text-decoration:none"><strong>Wilcox</strong></a>). Finally for both workflows, 100 runs with subsamples comprised of %60 of the original sample size were generated (without replacement)
        to validate significantly expressed genes and partially rule out sample bias (tagged <a class="text-warning" style="text-decoration:none"><strong>DESeq2 Validation</strong></a> and <a class="text-warning" style="text-decoration:none"><strong>Wilcox Validation</strong></a>).   
      </p>
    </div>

    <br>

    <div class="row">
      <h3 class="text-muted">Proteomics cross reference</h1>
      <p>
        Raw LC-MS/MS data of blood-brain barrier intracerebral microvessels (tagged "BBB") were downloaded from the PRIDE accession <a class="text-info" href="https://www.ebi.ac.uk/pride/archive/projects/PXD018602" target="_blank">PXD018602</a>. Protein intensities were quantified through <a class="text-info" href="https://www.maxquant.org/" target="_blank">MaxQuant</a> (ver. 2.3.0.0) with the provided PTMs.
        After filtering out potential contaminants and decoys, proteins with Q < 0.01 were cross checked with the main aggregate list. 
      </p>
    </div>

    <br>

    <div class="row">
      <h3 class="text-muted">Perfusion scores</h1>
      <p>
        Each significant appearance of a gene for liver, heart, skin, kidney and lung comparisons was given a perfusion score, which was obtained by calculating the perfusion rate (ml/100g/min) using <a class="text-info" href="https://bionumbers.hms.harvard.edu/files/Density%20and%20mass%20of%20each%20organ-tissue.pdf" target="_blank">tissue mass</a>  and <a class="text-info" href=" https://doi.org/10.2165/00003088-200847040-00004" target="_blank">blood flow (table 1)</a>. 
        The perfusion score is the mean of the sum of perfusion scores for all runs (4 in total).    
      </p>
    </div>

    <br>

    <div class="row">
      <h3 class="text-muted">Tables guide</h1>
      <p>
        Click the <a class="text-warning" style="text-decoration:none"><strong>yellow tags</strong></a> to switch between aggregated lists. Within a list, click the <a class="text-info" style="text-decoration:none"><strong>Gene symbol</strong></a> of an entry to search its corresponding
        UniProt ID over InterPro. To see the count distribution of samples over tissues, click the <a class="text-info" style="text-decoration:none"><strong>Counts</strong></a> button. For only brain endothelial cells, use the <a class="text-info" style="text-decoration:none"><strong>Brain Intravariance</strong></a> button instead. 
      </p>
    </div>
  
    <br>
    <br>

    <!-- Genetable -->
    <div class="row">
      <h1 class="text-primary">Gene Tables</h1>
      <br>
    </div>
    <br>
      <!-- Tab switch -->
    <div class="container-fluid">
      <ul class="nav nav-tabs" role="tablist">
        <li class="nav-item" role="presentation">
          <a class="nav-link active text-warning" data-bs-toggle="tab" href="#All" aria-selected="true" role="tab" tabindex="-1"><strong>All</strong></a>
        </li>
        <li class="nav-item" role="presentation">
          <a class="nav-link text-warning" data-bs-toggle="tab" href="#DESeq2" aria-selected="false" role="tab"><strong>DESeq2</strong></a>
        </li>
        <li class="nav-item" role="presentation">
          <a class="nav-link text-warning" data-bs-toggle="tab" href="#DESeq2-Valid" aria-selected="false" role="tab"><strong>DESeq2 Validation</strong></a>
        </li>
        <li class="nav-item" role="presentation">
          <a class="nav-link text-warning" data-bs-toggle="tab" href="#Wilcox" aria-selected="false" role="tab"><strong>Wilcox</strong></a>
        </li>
        <li class="nav-item" role="presentation">
          <a class="nav-link text-warning" data-bs-toggle="tab" href="#Wilcox-Valid" aria-selected="false" role="tab"><strong>Wilcox Validation</strong></a>
        </li>
      </ul>
      <br>
      <!-- Tables -->
      <div id="navTabContent" class="tab-content">
        
        <!-- All -->
        <div class="tab-pane fade active show" id="All" role="tabpanel">
          <div class="table-responsive">
            <table class="table table-hover center">
              <thead>
                <tr class="table-active bg-transparent">
                  <th scope="col"><p class="text-info">Ensembl ID</p></th>
                  <th scope="col"><p class="text-info">Gene Symbol</p></th>
                  <th scope="col"><p class="text-info">Rank p-val</p></th>
                  <th scope="col"><p class="text-info">Mean Perfusion Score</p></th>
                  <th scope="col"><p class="text-info">Top in DESeq2 Aggregate?</p></th>
                  <th scope="col"><p class="text-info">Top in DESeq2 Bootstrap?</p></th>
                  <th scope="col"><p class="text-info">Top in WRST Aggregate?</p></th>
                  <th scope="col"><p class="text-info">Top in WRST Bootstrap?</p></th>
                  <th scope="col"><p class="text-info">Protein Evidence?</p></th>
                  <th scope="col"><p class="text-info text-center">Actions</p></th>
                
                </tr>
              </thead>
              <tbody>
                {% for key, val in genes_df.iterrows() %}
                <tr class="table-active">
                  <th scope="row">{{ val['Name'] }}</th>
                  <td> <a href="https://www.ebi.ac.uk/interpro/search/text/{{ val['uniprot'] }}" target="_blank">{{ val['gene_names'] }}</a> </td> 
                  <td>{{ val['Score'] }}</td>
                  <td>{{ val['Mean Perfusion Score'] }}</td>

                  <!-- Main page -->
                  <td>
                    {% if val['DESeq2_Appeared'] %}
                    &#x2714;
                    {% else %} 
                    &#x2718;
                    {% endif %}
                  </td>
                
                  <td>
                    {% if val['DESeq2_Validated'] %}
                    &#x2714;
                    {% else %} 
                    &#x2718;
                    {% endif %}
                  </td>
                
                  <td>
                    {% if val['Wilcox_Appeared'] %}
                    &#x2714;
                    {% else %} 
                    &#x2718;
                    {% endif %}
                  </td>
                
                  <td>
                    {% if val['Wilcox_Validated'] %}
                    &#x2714;
                    {% else %} 
                    &#x2718;
                    {% endif %}
                  </td>
                
                  <td>
                    {% if val['Prot_Evidence'] %}
                    &#x2714;
                    {% else %} 
                    &#x2718;
                    {% endif %}
                  </td>
                
                  <td>
                    <a href="/omics/plots/counts/{{ val['Name'] }}" class="btn btn-primary" target="_blank">Counts</a>
                    <a href="/omics/plots/intravar/{{ val['Name'] }}" class="btn btn-primary" target="_blank">Brain Intravariance</a>
                  </td>
                </tr>
                {% endfor %}
              </tbody>
            </table>
          </div>
        </div>

        <!-- DESeq2 -->
        <div class="tab-pane fade" id="DESeq2" role="tabpanel">
          <div class="table-responsive">
            <table class="table table-hover center">
              <thead>
                <tr class="table-active bg-transparent">
                  <th scope="col"><p class="text-info">Ensembl ID</p></th>
                  <th scope="col"><p class="text-info">Gene Symbol</p></th>
                  <th scope="col"><p class="text-info">Rank p-val</p></th>
                  <th scope="col"><p class="text-info">Mean Perfusion Score</p></th>
                  <th scope="col"><p class="text-info">Top in Bootstrap?</p></th>
                  <th scope="col"><p class="text-info text-center">Actions</p></th>
                </tr>
              </thead>
              <tbody>
                {% for key, val in genes_deseq2_df.iterrows() %}
                <tr class="table-active">
                  <th scope="row">{{ val['Name'] }}</th>
                  <td> <a href="https://www.ebi.ac.uk/interpro/search/text/{{ val['uniprot'] }}" target="_blank">{{ val['gene_names'] }}</a> </td> 
                  <td>{{ val['Score'] }}</td>
                  <td>{{ val['Mean Perfusion Score'] }}</td>
                  <td>
                    {% if val['Validated'] %}
                    &#x2714;
                    {% else %} 
                    &#x2718;
                    {% endif %}
                  </td>
                  <td class="text-center">
                    <a href="/omics/plots/counts/{{ val['Name'] }}" class="btn btn-primary" target="_blank">Counts</a>
                    <a href="/omics/plots/intravar/{{ val['Name'] }}" class="btn btn-primary" target="_blank">Brain Intravariance</a>
                  </td>
                </tr>
                {% endfor %}
              </tbody>
            </table>
          </div>
        </div>

        <!-- DESeq2 Validation -->
        <div class="tab-pane fade" id="DESeq2-Valid" role="tabpanel">
          <div class="table-responsive">
            <table class="table table-hover center">
              <thead>
                <tr class="table-active bg-transparent">
                  <th scope="col"><p class="text-info">Ensembl ID</p></th>
                  <th scope="col"><p class="text-info">Gene Symbol</p></th>
                  <th scope="col"><p class="text-info">Rank p-val</p></th>
                  <th scope="col"><p class="text-info">Mean Perfusion Score</p></th>
                  <th scope="col"><p class="text-info">Top in DESeq2?</p></th>
                  <th scope="col"><p class="text-info text-center">Actions</p></th>
                </tr>
              </thead>
              <tbody>
                {% for key, val in genes_deseq2_valid_df.iterrows() %}
                <tr class="table-active">
                  <th scope="row">{{ val['Name'] }}</th>
                  <td> <a href="https://www.ebi.ac.uk/interpro/search/text/{{ val['uniprot'] }}" target="_blank">{{ val['gene_names'] }}</a> </td> 
                  <td>{{ val['Score'] }}</td>
                  <td>{{ val['Mean Perfusion Score'] }}</td>
                  <td>
                    {% if val['Validated'] %}
                    &#x2714;
                    {% else %} 
                    &#x2718;
                    {% endif %}
                  </td>
                  <td class="text-center">
                    <a href="/omics/plots/counts/{{ val['Name'] }}" class="btn btn-primary" target="_blank">Counts</a>
                    <a href="/omics/plots/intravar/{{ val['Name'] }}" class="btn btn-primary" target="_blank">Brain Intravariance</a>
                  </td>
                </tr>
                {% endfor %}
              </tbody>
            </table>
          </div>
        </div>

        <!-- Wilcox -->
        <div class="tab-pane fade" id="Wilcox" role="tabpanel">
          <div class="table-responsive">
            <table class="table table-hover center">
              <thead>
                <tr class="table-active bg-transparent">
                  <th scope="col"><p class="text-info">Ensembl ID</p></th>
                  <th scope="col"><p class="text-info">Gene Symbol</p></th>
                  <th scope="col"><p class="text-info">Rank p-val</p></th>
                  <th scope="col"><p class="text-info">Mean Perfusion Score</p></th>
                  <th scope="col"><p class="text-info">Top in Bootstrap?</p></th>
                  <th scope="col"><p class="text-info text-center">Actions</p></th>
                </tr>
              </thead>
              <tbody>
                {% for key, val in genes_wilcox_df.iterrows() %}
                <tr class="table-active">
                  <th scope="row">{{ val['Name'] }}</th>
                  <td> <a href="https://www.ebi.ac.uk/interpro/search/text/{{ val['uniprot'] }}" target="_blank">{{ val['gene_names'] }}</a> </td> 
                  <td>{{ val['Score'] }}</td>
                  <td>{{ val['Mean Perfusion Score'] }}</td>
                  <td>
                    {% if val['Validated'] %}
                    &#x2714;
                    {% else %} 
                    &#x2718;
                    {% endif %}
                  </td>
                  <td class="text-center">
                    <a href="/omics/plots/counts/{{ val['Name'] }}" class="btn btn-primary" target="_blank">Counts</a>
                    <a href="/omics/plots/intravar/{{ val['Name'] }}" class="btn btn-primary" target="_blank">Brain Intravariance</a>
                  </td>
                </tr>
                {% endfor %}
              </tbody>
            </table>
          </div>
        </div>

        <!-- Wilcox Validation -->
        <div class="tab-pane fade" id="Wilcox-Valid" role="tabpanel">
          <div class="table-responsive">
            <table class="table table-hover center">
              <thead>
                <tr class="table-active bg-transparent">
                  <th scope="col"><p class="text-info">Ensembl ID</p></th>
                  <th scope="col"><p class="text-info">Gene Symbol</p></th>
                  <th scope="col"><p class="text-info">Rank p-val</p></th>
                  <th scope="col"><p class="text-info">Mean Perfusion Score</p></th>
                  <th scope="col"><p class="text-info">Top in Wilcox?</p></th>
                  <th scope="col"><p class="text-info text-center">Actions</p></th>
                </tr>
              </thead>
              <tbody>
                {% for key, val in genes_wilcox_valid_df.iterrows() %}
                <tr class="table-active">
                  <th scope="row">{{ val['Name'] }}</th>
                  <td> <a href="https://www.ebi.ac.uk/interpro/search/text/{{ val['uniprot'] }}" target="_blank">{{ val['gene_names'] }}</a> </td> 
                  <td>{{ val['Score'] }}</td>
                  <td>{{ val['Mean Perfusion Score'] }}</td>
                  <td>
                    {% if val['Validated'] %}
                    &#x2714;
                    {% else %} 
                    &#x2718;
                    {% endif %}
                  </td>
                  <td class="text-center">
                    <a href="/omics/plots/counts/{{ val['Name'] }}" class="btn btn-primary" target="_blank">Counts</a>
                    <a href="/omics/plots/intravar/{{ val['Name'] }}" class="btn btn-primary" target="_blank">Brain Intravariance</a>
                  </td>
                </tr>
                {% endfor %}
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
    
  </div>

    
  <!-- Footer -->
  <footer id="footer" class="container-fluid p-4">
    <ul class="list-group list-group-horizontal">
      <li class="list-group-item bg-transparent"><a href="">GitHub</a></li>
      <li class="list-group-item bg-transparent"><a href="">Links-1</a></li>
      <li class="list-group-item bg-transparent"><a href="">Links-2</a></li>
    </ul>
    <br>
    <div class="row">
      <p>License info...</p>
      <p>Acknowledgements...</p>
    </div>
  </footer>
<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/js/bootstrap.bundle.min.js" integrity="sha384-MrcW6ZMFYlzcLA8Nl+NtUVF0sA7MsXsP1UyJoMp4YLEuNSfAP+JcXn/tWtIaxVXM" crossorigin="anonymous"></script>
</body>
</html>